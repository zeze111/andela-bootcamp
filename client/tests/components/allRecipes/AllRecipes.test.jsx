import React from 'react';

import connectedAllRecipes, {AllRecipes} from '../../../components/allRecipes/AllRecipes';
import store from '../../../store';

const props = {
  getPaginatedRecipes: jest.fn(() => Promise.resolve()),
  getRecipeCategory: jest.fn(() => Promise.resolve()),
  searchRecipe: jest.fn(() => Promise.resolve()),
  recipes: [
    {
      id: 1,
      name: 'something'
    },
    {
      id: 2,
      name: 'something'
    },
  ]
};

const setup = () => {
  const wrapper = shallow(<AllRecipes {...props} store={store} />);
  return { wrapper }
}

describe('component will mount', () => {
  it('should call get recipes function', () => {
    const spy = sinon.spy(AllRecipes.prototype, 'componentWillMount');
  });
});

describe('onNextPage', () => {
  it('is called when next/previous page is called', () => {
    const { wrapper } = setup();
    const event = 1;

    wrapper.instance().onNextPage(event);
  });
});

describe('onNextPage', () => {
  it('is called when next/previous page is called for category', () => {
    const { wrapper } = setup();
    const event = 1;

    wrapper.setState({
      dropdown: 'Main'
    });

    wrapper.instance().onNextPage(event);
  });
});

describe('onNextPage', () => {
  it('is called when next/previous page is called for search', () => {
    const { wrapper } = setup();
    const event = 1;

    wrapper.setState({
      searching: true
    });

    wrapper.instance().onNextPage(event);
  });
});

describe('onSelectCategory', () => {
  it('is called when a category is selected', () => {
    const { wrapper } = setup();
    const event = 'Main';

    wrapper.instance().onSelectCategory(event);
  });
});

describe('onSelectAllRecipes', () => {
  it('is called when All recipes category is selected', () => {
    const { wrapper } = setup();
    const event = 'All Recipes';

    wrapper.instance().onSelectAllRecipes(event);
  });
});

describe('onChange', () => {
  it('is called when searching for a recipe', () => {
    const { wrapper } = setup();
    const event = {
      target: { value: 'rice' }
    };
    const search = wrapper.find('#search');

    wrapper.instance().onChange(event);
  });
});

describe('onChange', () => {
  it('is called when searching for a recipe', () => {
    const { wrapper } = setup();
    const event = {
      target: { value: '' }
    };
    const search = wrapper.find('#search');

    wrapper.instance().onChange(event);
  });
});

describe('Connect All Recipes component', () => {
  it('should render component successfully', () => {
    const store = mockStore({
        recipeReducer: {
          recipes: [],
          message: '',
          pagination: {}
        }
    })

    const wrapper = shallow(<AllRecipes {...props} store={store} />);
    expect(wrapper.length).toBe(1);
  });
});

describe('Test for All Recipes', () => {
  it('should render component', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot()
  });
});
