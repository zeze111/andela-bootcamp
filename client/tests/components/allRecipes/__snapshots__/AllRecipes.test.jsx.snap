// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for All Recipes should render component 1`] = `
Object {
  "wrapper": <main>
    <div
      className="container full-container"
    >
      <div
        className="row paginate"
      >
        <div
          className="col l5 m7 s10 push-l2"
        >
          <Categories
            dropdown="All Recipes"
            onSelectAllRecipes={[Function]}
            onSelectCategory={[Function]}
          />
        </div>
        <Search
          onChange={[Function]}
          search=""
        />
      </div>
    </div>
    <div
      className="row remove-margin-bottom home-width text3"
    >
      <div
        className="center-align loader-style min-preloader"
      >
        <PreLoader />
      </div>
      <ul
        className="categories all-style flex-container-homepage"
      >
        <RecipeCard
          key="1"
          recipe={
            Object {
              "id": 1,
              "name": "something",
            }
          }
        />
        <RecipeCard
          key="2"
          recipe={
            Object {
              "id": 2,
              "name": "something",
            }
          }
        />
      </ul>
      <div
        className="row paginate"
      >
        <div
          className="center-align"
        >
          <Pagination
            activePage={1}
            items={0}
            onSelect={[Function]}
          />
        </div>
      </div>
    </div>
  </main>,
}
`;
